05/rodne_cislo_finish.py

```diff
@@ -0,0 +1,60 @@
+# Input ID number in Czech Republic
+# write ouf corect/incorect ID
+
+
+# MONTHS 30 DAYS
+months_30 = [4, 6, 9, 11]
+
+# MONTHS 31 DAYS
+months_31 = [1, 3, 5, 7, 8, 10, 12]
+
+# YEARS
+years = list(range(86, 100)) + list(range(00, 23))
+only_numbers = []
+
+while True:
+
+    ID = input('Napiš rodné číslo: ')
+
+    if ID[6] != "/":
+        print("chybí lomítko nebo je špatný počet čísel")
+        continue
+
+    for number in ID:
```

Comment:
Tento for cyklus je trochu zradný. Ak na vstupe zadáš napr. `123456/1234.` tak sa zacyklíš (program sa znova spýta na vstup) ale nespadne to na žiadnej chybe. Lepšie je skontrolovať si dĺžku celého inputu (11), a jednoducho si ho rozdeliť na prvých 6 znakov a posledné 4 a re časti si overiť či je to číslo (.isdigt()). Takto vlastne z inputu vytiahneš len čísla a ak sa tam zatúla nejaký znak navyše (nie číslo) tak ho preskočíš, pričom by to malo hlásič chybu.

Úplne stačí takto: 
```
ID = input('Napiš rodné číslo: ')
if len(ID) != 11:
    print("Invalid lenght.")
    continue

number_digits = (number.replace('/',''))
if not number_digits.isdigit() and ID[6] != "/":
    print("Is not in proper format")
    continue
```

Potom už nemusíš riešiš spájanie listu, na riadkoch 32-36. Stačí si ho jednorazovo previesť na číslo a potom už len berieš tie časti ktoré chceš.